nym-node-status-api/nym-node-status-api/src/monitor/mod.rs

```diff
@@ -70,345 +85,342 @@ pub(crate) async fn spawn_in_background(
+            .log_error("get_active_mixnodes")?;
+        let delegation_program_members =
+            get_delegation_program_details(&self.network_details, &self.nyxd_addr).await?;
+
+        // keep stats for later
+        let count_bonded_mixnodes = mixnodes.len();
+        let count_bonded_gateways = gateways.len();
+        let count_bonded_mixnodes_active = mixnodes_active.len();
+
+        let gateway_records = self.prepare_gateway_data(
+            &gateways,
+            &gateways_blacklisted,
+            gateway_geodata,
+            all_skimmed_nodes,
+        )?;
+
+        let pool = self.db_pool.clone();
+        queries::insert_gateways(&pool, gateway_records)
+            .await
+            .map(|_| {
+                tracing::debug!("Gateway info written to DB!");
+            })?;
 
-    // let gateways_blacklisted = gateways.iter().filter(|gw|gw.)
-    let gateways_blacklisted = all_skimmed_nodes
-        .iter()
-        .filter_map(|node| {
-            if node.performance.round_to_integer() <= 50 && node.supported_roles.entry {
-                Some(node.ed25519_identity_pubkey.to_base58_string())
-            } else {
-                None
-            }
-        })
-        .collect::<HashSet<_>>();
-
-    // Cached mixnodes don't include blacklisted nodes
-    // We need that to calculate the total locked tokens later
-    let mixnodes = api_client
-        .nym_api
-        .get_mixnodes_detailed_unfiltered()
-        .await
-        .log_error("get_mixnodes_detailed_unfiltered")?;
-    let mixnodes_described = api_client
-        .nym_api
-        .get_mixnodes_described()
-        .await
-        .log_error("get_mixnodes_described")?;
-    let mixnodes_active = api_client
-        .nym_api
-        .get_active_mixnodes()
-        .await
-        .log_error("get_active_mixnodes")?;
-    let delegation_program_members =
-        get_delegation_program_details(network_details, nyxd_addr).await?;
-
-    // keep stats for later
-    let count_bonded_mixnodes = mixnodes.len();
-    let count_bonded_gateways = gateways.len();
-    let count_explorer_gateways = explorer_gateways.len();
-    let count_bonded_mixnodes_active = mixnodes_active.len();
-
-    let gateway_records = prepare_gateway_data(
-        &gateways,
-        &gateways_blacklisted,
-        explorer_gateways,
-        all_skimmed_nodes,
-    )?;
-    queries::insert_gateways(pool, gateway_records)
-        .await
-        .map(|_| {
-            tracing::debug!("Gateway info written to DB!");
-        })?;
-
-    // instead of counting blacklisted GWs returned from API cache, count from the active set
-    let count_gateways_blacklisted = gateways
-        .iter()
-        .filter(|gw| {
-            let gw_identity = gw.ed25519_identity_key().to_base58_string();
-            gateways_blacklisted.contains(&gw_identity)
-        })
-        .count();
+        // instead of counting blacklisted GWs returned from API cache, count from the active set
```

Comment:
active set? but you never queried for active set gateways. I'm a bit confused here